controller/kubernetes_controller.go

```diff
@@ -264,27 +269,22 @@ func (kc *KubernetesController) syncKubernetesStatus(key string) (err error) {
 			// The associated PVC is removed from the PV ClaimRef
 			if ks.PVCName != "" {
 				ks.LastPVCRefAt = kc.nowHandler()
+				if len(ks.WorkloadsStatus) != 0 && ks.LastPodRefAt == "" {
+					ks.LastPodRefAt = kc.nowHandler()
+				}
 			}
 		}
 	}
 
-	pod, err := kc.getAssociatedPod(ks.Namespace, ks.PVCName)
+	// podNum includes terminating pods
+	pods, podNum, err := kc.getAssociatedPods(ks)
```

Comment:
The name `podNum` is too vague. It's better to change it to `terminatingPodCount` and exclude the ones already in the `pods`. Then in `cleanupVolumeAttachment` you only need to check `if terminatingPodCount == 0`